shared/src/main/scala/tastyquery/reader/PositionUnpickler.scala

```diff
@@ -0,0 +1,76 @@
+/** Modified from dotty.tools.dotc.core.tasty.PositionUnpickler */
+
+package tastyquery.reader
+
+/** Uses the TermName defined in tastyquery */
+import tastyquery.ast.Names.TermName
+import tastyquery.ast.Spans.{Span, NoSpan}
+import tastyquery.util
+
+import dotty.tools.tasty.{TastyFormat, TastyBuffer, TastyReader}
+import TastyFormat.SOURCE
+import TastyBuffer.{Addr, NameRef}
+
+/** Unpickler for tree positions */
+class PositionUnpickler(reader: TastyReader, nameAtRef: NameRef => TermName) {
+  import reader.*
+
+  private var myLineSizes: Array[Int] = _
+  private var mySpans: util.HashMap[Addr, Span] = _
```

Comment:
I think it is fine to use standard hash maps from the std library for now, if we identify the need we can optimise later

Reply:
Done